src/appleseed/renderer/modeling/input/source.h

```diff
@@ -94,6 +94,34 @@ class APPLESEED_DLLSYMBOL Source
         Spectrum&                   spectrum,
         Alpha&                      alpha) const;
 
+    // Evaluate the source at a given shading point.
+    virtual void evaluate(
```

Comment:
You need to modify the existing methods to replace the uv argument by a ShadingPoint, not adding new ones.